sdk/src/main/java/com/deploygate/sdk/DeployGate.java

```diff
@@ -828,16 +998,38 @@ private void registerCallbackInternal(
      *         callback listener to be removed
      *
      * @since r1
+     * @deprecated since 4.9.0. Use each unregister method for separated callbacks instead.
+     * @see #unregisterInitializeCallback(DeployGateInitializeCallback)
+     * @see #unregisterStatusChangeCallback(DeployGateStatusChangeCallback)
+     * @see #unregisterUpdateAvailableCallback(DeployGateUpdateAvailableCallback)
      */
-    public static void unregisterCallback(DeployGateCallback listener) {
+    @Deprecated
+    public static void unregisterCallback(final DeployGateCallback listener) {
         if (sInstance == null) {
             return;
         }
         if (listener == null) {
             return;
         }
 
-        sInstance.mCallbacks.remove(listener);
+        sInstance.mInitializeCallbacks.remove(new DeployGateInitializeCallback() {
```

Comment:
These codes are completely wrong. Please re-check what this code expects.

By the way, the test job passed so the test codes of `unregister` methods are also broken or not tested. Could you please fix them as well?

Reply:
No testcase for register/unregister methods exists.
Added it and fixed my code causing unexpected behavior.